client/modules/core/main.js

```diff
@@ -180,60 +180,38 @@ export default {
     let permissions = ["owner"];
     let id = "";
     const userId = checkUserId || Meteor.userId();
-    //
-    // local roleCheck function
-    // is the bulk of the logic
-    // called out a userId is validated.
-    //
+
+    // This is declared as a function for the sake of reuse. Note that it is
+    // only called after userId is validated.
     function roleCheck() {
       // permissions can be either a string or an array
       // we'll force it into an array and use that
+      // TODO: Should we also cater for checkPermissions === "" and checkPermissions === []?
```

Comment:
It doesn't hurt, but it can also be deferred to final Roles method check just like this other [comment](https://github.com/reactioncommerce/reaction/pull/3196#discussion_r148066088)

Reply:
Noted.